contracts/staking.tmy/src/staking.tmy.cpp

```diff
@@ -239,6 +239,11 @@ namespace stakingtoken
             });
 
             total_yield += yield;
+         } 
+         else if (itr->unstake_requested && eosio::current_time_point() >= itr->unstake_time + eosio::days(5) ) 
+         {
+            // If user has requested unstake, check if 5 days have passed since unstake request
+            releasetoken(staker, itr->id);
```

Comment:
this wont work, as `releasetoken()` does a `require_auth()` check that will fail. you should make a private function without the auth check which is called here

the action ``releasetoken()` will then call that private function after calling `require_auth()`